requirements_test.txt

```diff
@@ -0,0 +1,10 @@
+-r requirements_test_pre_commit.txt
```

Comment:
This Is not required, pre-commit handles its dependencies itself in à virtualenv (except pylint because its a systeme hook)

Reply:
But we could include `requirements_test_pypy.txt ` instead to avoid duplication ?

Reply:
To include `requirements_test_pypy.txt` is a good idea. I changed that.

Regarding `requirements_test_pre_commit.txt`: I'm aware that those dependencies aren't required strictly speaking. That's why I'm not including it in the hash calculation. So if that file changes, it won't recreate the python cache because of it. My idea was to allow developers to use
```bash
python -m venv venv
source venv/bin/activate
pip install -r requirements_test.txt
```
to setup their complete dev environment. That should include the packages used by pre-commit since some IDEs use them to add additional features, eg. VS Code together with black can support `Format on save`. It also allow devs to run checks manually (mypy). The only small downside I see is that we need to make sure to keep the versions in sync.

Reply:
Yeah, we had a discrepancy in version at some point that was causing problems. I think being able to launch tools directly is great but wouldn't it be better to let the user install the linters themself at the proper version defined in the pre-commit configuration ?

Reply:
Let's assume the versions are updated together and match. The user would then only need to run
```
pip install -r requirements_test.txt
```
to install the linters in the correct version.

The alternative AFAIK would be to open `.pre-commit-config` and look for each installed check. I don't assume that everyone is that familiar with pre-commit.

We could argue to include the `requirements_test_pre_commit.txt` file, but remove it from `requirements_test.txt`. However, I guess many do use linters (black, isort, flake) outside of pre-commit and what really is the downside in including it anyway?

Reply:
Ok let's do this.